src/main/java/uk/gov/register/core/PostgresRegister.java

```diff
@@ -20,107 +20,113 @@
     }
 
     @Override
     public Optional<Item> getItemBySha256(String sha256hex) {
-        return registerDAO.getItemBySha256(sha256hex);
+        return dbi.withHandle(h -> itemStore.getItemBySha256(h, sha256hex));
     }
 
     @Override
     public int getTotalEntries() {
-        return registerDAO.getTotalEntries();
+        return dbi.withHandle(entryLog::getTotalEntries);
     }
 
     @Override
     public Collection<Entry> getEntries(int start, int limit) {
-        return registerDAO.getEntries(start, limit);
+        return dbi.withHandle(h -> entryLog.getEntries(h, start, limit));
     }
 
     @Override
     public Optional<Record> getRecord(String key) {
-        return registerDAO.getRecord(key);
+        return dbi.withHandle(h -> recordIndex.getRecord(h, key));
     }
 
     @Override
     public Collection<Entry> allEntriesOfRecord(String key) {
-        return registerDAO.findAllEntriesOfRecordBy(registerName, key);
+        return dbi.withHandle(h -> recordIndex.findAllEntriesOfRecordBy(h, registerName, key));
     }
 
     @Override
     public List<Record> getRecords(int limit, int offset) {
-        return registerDAO.getRecords(limit, offset);
+        return dbi.withHandle(h -> recordIndex.getRecords(h, limit, offset));
     }
 
     @Override
     public Collection<Entry> getAllEntries() {
-        return registerDAO.getAllEntries();
+        return dbi.withHandle(entryLog::getAllEntries);
     }
 
     @Override
     public Collection<Item> getAllItems() {
-        return registerDAO.getAllItems();
+        return dbi.withHandle(itemStore::getAllItems);
     }
 
     @Override
     public int getTotalRecords() {
-        return registerDAO.getTotalRecords();
+        return dbi.withHandle(recordIndex::getTotalRecords);
     }
 
     @Override
     public Optional<Instant> getLastUpdatedTime() {
-        return registerDAO.getLastUpdatedTime();
+        return dbi.withHandle(entryLog::getLastUpdatedTime);
     }
 
     @Override
     public List<Record> max100RecordsFacetedByKeyValue(String key, String value) {
-        return registerDAO.findMax100RecordsByKeyValue(key, value);
+        return dbi.withHandle(h -> recordIndex.findMax100RecordsByKeyValue(h, key, value));
     }
 
     @Override
     public RegisterProof getRegisterProof() throws NoSuchAlgorithmException {
-        return verifiableLogService.getRegisterProof();
+        return dbi.withHandle(entryLog::getRegisterProof);
     }
 
     @Override
     public EntryProof getEntryProof(int entryNumber, int totalEntries) {
-        return verifiableLogService.getEntryProof(entryNumber, totalEntries);
+        return dbi.withHandle(handle ->
+                entryLog.getEntryProof(handle, entryNumber, totalEntries));
     }
 
     @Override
     public ConsistencyProof getConsistencyProof(int totalEntries1, int totalEntries2) {
-        return verifiableLogService.getConsistencyProof(totalEntries1, totalEntries2);
+        return dbi.withHandle(handle ->
```

Comment:
We'll want to double check that this is wrapped in a transaction, or explicitly open a transaction